spark/src/main/scala/io/delta/sharing/spark/filters/JsonPredicates.scala

```diff
@@ -0,0 +1,346 @@
+    ctx.partitionValues.getOrElse(name, null)
+  }
+}
+
+/**
+ * Represents a literal.
+ *
+ * @param value The value of this literal as a string.
+ * @param valueType The type of value.
+ */
+case class LiteralOp(val value: String, val valueType: String) extends LeafOp {
+
+  override def validate(): Unit = {
+    if (value == null) {
+      throw new IllegalArgumentException("Value must be specified: " + this)
+    }
+    if (!OpDataTypes.supportedTypes.contains(valueType)) {
+      throw new IllegalArgumentException("Unsupported type: " + valueType)
+    }
+  }
+
+  // Returns the value and its type.
+  override def eval(ctx: EvalContext): Any = (value, valueType)
+
+  // Literals are always non-null.
+  override def isNull(ctx: EvalContext): Boolean = false
+}
+
+/**
+ * Represents a null operation on its child.
+ */
+case class IsNullOp(children: Seq[LeafOp]) extends NonLeafOp with UnaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any = children(0).isNull(ctx)
+}
+
+/**
+ * The following ops represent comparison operations on their children.
+ * The (logically) supported operations are:
+ *   "=", "<", "<=", ">", ">="
+ *
+ * We only implement "=" and "<", and represent all others in terms
+ * of these two.
+ *
+ * @param children Expected size is 2.
+ */
+
+case class EqualOp(children: Seq[LeafOp]) extends NonLeafOp with BinaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any = EvalHelper.equal(children, ctx)
+}
+
+case class LessThanOp(children: Seq[LeafOp]) extends NonLeafOp with BinaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any = EvalHelper.lessThan(children, ctx)
+}
+
+case class LessThanOrEqualOp(children: Seq[LeafOp]) extends NonLeafOp with BinaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any =
+    EvalHelper.lessThan(children, ctx) || EvalHelper.equal(children, ctx)
+}
+
+case class GreaterThanOp(children: Seq[LeafOp]) extends NonLeafOp with BinaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any =
+    !EvalHelper.lessThan(children, ctx) && !EvalHelper.equal(children, ctx)
+}
+
+case class GreaterThanOrEqualOp(children: Seq[LeafOp]) extends NonLeafOp with BinaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any = !EvalHelper.lessThan(children, ctx)
+}
+
+/**
+ * The following ops represent boolean logic related operations.
+ * The (logically) supported operations are:
+ *   "&&", "||", "!"
+ */
+
+case class AndOp(val children: Seq[NonLeafOp]) extends NonLeafOp with NaryOp {
+  override def validate(): Unit = validateChildren(children)
+
+  override def eval(ctx: EvalContext): Any = {
+    children.forall(c => c.evalExpectBoolean(ctx))
```

Comment:
This assumes the children are all non-leaf node. What about expression like `True AND False`? I guess it's not supported.

Reply:
This implementation is intentionally restricted towards file filtering based on partition values and stats.
 So we don't expect to compare two leaf level literals. 
The "True" node will typically be an output of either a comparator node, or a logical operation.

These constraints are very easy to relax if we see a use case.

Reply:
Made the children baseOps